kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/model/ingress/ClusterIPIngressDefinitionSniHostIdentifiesNodeTest.java

```diff
@@ -0,0 +1,371 @@
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyBuilder;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyIngress;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyIngressBuilder;
+import io.kroxylicious.kubernetes.api.v1alpha1.VirtualKafkaCluster;
+import io.kroxylicious.kubernetes.api.v1alpha1.VirtualKafkaClusterBuilder;
+import io.kroxylicious.kubernetes.api.v1alpha1.kafkaproxyingressspec.ClusterIP;
+import io.kroxylicious.kubernetes.api.v1alpha1.kafkaservicespec.NodeIdRanges;
+import io.kroxylicious.kubernetes.api.v1alpha1.kafkaservicespec.NodeIdRangesBuilder;
+import io.kroxylicious.kubernetes.api.v1alpha1.virtualkafkaclusterspec.ingresses.Tls;
+import io.kroxylicious.kubernetes.api.v1alpha1.virtualkafkaclusterspec.ingresses.TlsBuilder;
+import io.kroxylicious.kubernetes.operator.KafkaProxyReconciler;
+import io.kroxylicious.kubernetes.operator.ProxyDeploymentDependentResource;
+import io.kroxylicious.proxy.config.VirtualClusterGateway;
+import io.kroxylicious.proxy.service.HostPort;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.params.provider.Arguments.argumentSet;
+
+class ClusterIPIngressDefinitionSniHostIdentifiesNodeTest {
+
+    private static final String INGRESS_NAME = "my-ingress";
+    public static final String CLUSTER_NAME = "my-cluster";
+    public static final String PROXY_NAME = "my-proxy";
+    public static final String NAMESPACE = "my-namespace";
+
+    // @formatter:off
+    private static final KafkaProxyIngress INGRESS = new KafkaProxyIngressBuilder()
+            .withNewMetadata()
+                .withName(INGRESS_NAME)
+                .withNamespace(NAMESPACE)
+            .endMetadata()
+            .withNewSpec()
+                .withNewClusterIP()
+                    .withProtocol(ClusterIP.Protocol.TLS)
+                    .withNodeIdentifiedBy(ClusterIP.NodeIdentifiedBy.SNI)
+                .endClusterIP()
+            .endSpec()
+            .build();
+    // @formatter:on
+
+    public static final String CERTIFICATE_NAME = "certificate";
+    private static final VirtualKafkaCluster VIRTUAL_KAFKA_CLUSTER = new VirtualKafkaClusterBuilder()
+            .withNewMetadata()
+            .withName(CLUSTER_NAME)
+            .withNamespace(NAMESPACE)
+            .endMetadata()
+            .withNewSpec()
+            .addNewIngress()
+            .withNewIngressRef()
+            .withName(INGRESS_NAME)
+            .endIngressRef()
+            .withNewTls()
+            .withNewCertificateRef()
+            .withName(CERTIFICATE_NAME)
+            .endCertificateRef()
+            .endTls()
+            .endIngress()
+            .endSpec()
+            .build();
+    // @formatter:on
```

Comment:
we haven't turned the formatter off prior to this and nor have we done anything that it would touch (but maybe thats because its on already?)